### PR TITLE
 Pull the latest tagged release

### PR DESCRIPTION
**Pull request by mxr576**

### CODE REVIEW

File: scripts/devportal/update-themes-and-modules.sh [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
 declare -r THEMES_DIR='web/themes/devportal'
 declare -a modules
 declare -a themes
+declare -r DEVPORTAL_REMOTE_REPOSITORY_URL=${DEVPORTAL_REMOTE_REPOSITORY_URL-:"git@bitbucket.org:pronovix"}
 
-for dir in ./${MODULES_DIR}/*/; do
-    modules+=("$(basename "${dir}")")
-done
+if [[ -d ${MODULES_DIR} ]]; then
+  for dir in $(find ${MODULES_DIR} -mindepth 1 -maxdepth 1 -type d); do
```
Comment (mxr576) [inline scripts/devportal/update-themes-and-modules.sh:14]: I had some issues with running the original script on Linux this is the reason why I changed.

File: scripts/devportal/update-themes-and-modules.sh [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
-# UPDATE ALL PRONOVIX MODULES
+# update-themes-and-modules.sh
+# UPDATE ALL PRONOVIX MODULES AND THEMES.
 
 # ----- CONSTANTS ------
 declare -r MODULES_DIR='web/modules/devportal'
 declare -r THEMES_DIR='web/themes/devportal'
 declare -a modules
 declare -a themes
+declare -r DEVPORTAL_REMOTE_REPOSITORY_URL=${DEVPORTAL_REMOTE_REPOSITORY_URL-:"git@bitbucket.org:pronovix"}
```
Comment (mxr576) [inline scripts/devportal/update-themes-and-modules.sh:11]: I have two BB account registered with different SSH keys.

File: scripts/devportal/update-themes-and-modules.sh [hunk 3]
```diff
@@ -25,33 +30,53 @@ function add_all_repositories {
 function add_all_modules_repositories {
     for repo_name in ${modules[@]}
     do
-        git remote add -f ${repo_name} git@bitbucket.org:pronovix/${repo_name}.git
+      register_repository ${repo_name}
```
Comment (boobaa) [inline scripts/devportal/update-themes-and-modules.sh]: If I'm not mistaken, you're assuming that the directory names in `web/modules/devportal/` are repo names – which is merely a coincidence. Could you pls also document this need?

File: scripts/devportal/update-themes-and-modules.sh [hunk 4]
```diff
@@ -25,33 +30,53 @@ function add_all_repositories {
...
     done
 }
 
 function add_all_themes_repositories {
     for repo_name in ${themes[@]}
     do
-        git remote add -f ${repo_name} git@bitbucket.org:pronovix/${repo_name}.git
+      register_repository ${repo_name}
```
Comment (boobaa) [inline scripts/devportal/update-themes-and-modules.sh]: See my above comment regarding modules; this is the same about themes this time.

File: scripts/devportal/update-themes-and-modules.sh [hunk 5]
```diff
@@ -25,33 +30,53 @@ function add_all_repositories {
...
     done
 }
 
+function register_repository {
+  local repo_name=$1
+  git remote get-url ${repo_name} &> /dev/null
+  if [[ $? -gt 0 ]]; then
+    git remote add -f ${repo_name} ${DEVPORTAL_REMOTE_REPOSITORY_URL}/${repo_name}.git
+  fi
```
Comment (boobaa) [inline scripts/devportal/update-themes-and-modules.sh:50]: So what's gonna happen if the directory name is NOT a repo name? Will this directory be skipped silently?

**Review by boobaa — CHANGES_REQUESTED** [review]
Oh, and please be consistent with the indents: I'm not telling it should be 2 or 4 spaces, but it should be consistent.

File: scripts/devportal/update-themes-and-modules.sh (hunk 3 above)
Comment (mxr576) [reply scripts/devportal/update-themes-and-modules.sh]: Yep, you are right, this requirement is coming from the original code. It was also weird for me, but with this approach at least we do not expose our repo names.

File: scripts/devportal/update-themes-and-modules.sh (hunk 5 above)
Comment (mxr576) [reply scripts/devportal/update-themes-and-modules.sh:50]: You get an error from GIT that the repository does not exist but the script continues to run.

File: scripts/devportal/update-themes-and-modules.sh (hunk 3 above)
Comment (mxr576) [reply scripts/devportal/update-themes-and-modules.sh]: This is already documented here: https://github.com/Pronovix/devportal-starterkit#updating-devportal-themes-and-modules

**Pull request closed without merging**